discord/interactions.py

```diff
@@ -624,6 +638,94 @@ async def translate(
         return await translator.translate(string, locale=locale, context=context)
 
 
+class InteractionCallbackActivityInstance:
+    """Represents an activity instance launched as an interaction response.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    id: :class:`str`
+        The activity instance ID.
+    """
+
+    __slots__ = ('id',)
+
+    def __init__(self, data: InteractionCallbackActivityPayload) -> None:
+        self.id: str = data['id']
+
+
+class InteractionCallback(Generic[ClientT]):
+    """Represents an interaction response callback.
+
+    .. versionadded:: 2.5
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The interaction ID.
+    type: :class:`InteractionResponseType`
+        The interaction callback response type.
+    resource: Optional[Union[:class:`InteractionMessage`, :class:`InteractionCallbackActivityInstance`]]
+        The resource that the interaction response created. If a message was sent, this will be
+        a :class:`InteractionMessage`, else if an activity was launched this will be a
+        :class:`InteractionCallbackActivityInstance`. In any other case, this will be ``None``.
```

Comment:
```suggestion
        The resource that the interaction response created. If a message was sent, this will be
        a :class:`InteractionMessage`. If an activity was launched this will be a
        :class:`InteractionCallbackActivityInstance`. In any other case, this will be ``None``.
```